app/src/main/java/com/omkarmoghe/pokemap/views/map/MapWrapperFragment.java

```diff
@@ -596,12 +596,12 @@ public void onAnimationStart(Animator animator) {
 
             @Override
             public void onAnimationEnd(Animator animator) {
+                pokemonMarker.getMarker().remove();
+                markerList.remove(pokemonMarker);
             }
 
             @Override
             public void onAnimationCancel(Animator animator) {
-                pokemonMarker.getMarker().remove();
```

Comment:
Thank you @rohanjain1 and @s7092910 , I just updated the code, handle both onAnimationEnd and onAnimationCancel.